### PR TITLE
feat(safe-mode): Empty instead of undefined

### PR DESCRIPTION
**Pull request by brandonseydel**
# Pull Request

## 📖 Description
This makes a default "safe" mode for aurelia that will print empty strings instead of undefined.
```ts
class {test}
```
```html
<div>${'1' + test}</div>
<!--strict mode = '1undefined' -->
<!--safe mode = '1' -->
```

## 📑 Test Plan
Need to create more tests for switching the flag back and forth from safe to strict.

### CODE REVIEW

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://30307-121594173-gh.circle-artifacts.com/0/home/circleci/repo/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://30333-121594173-gh.circle-artifacts.com/0/home/circleci/repo/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://30403-121594173-gh.circle-artifacts.com/0/home/circleci/repo/test/js-framework-benchmark/webdriver-ts-results/table.html

File: packages/runtime/src/flags.ts [hunk 1]
```diff
@@ -105,6 +105,7 @@ export const enum LifecycleFlags {
   bindingStrategy               = 0b00000_0000000_0000000000000_00_1111,
   getterSetterStrategy          = 0b00000_0000000_0000000000000_00_0001,
   proxyStrategy                 = 0b00000_0000000_0000000000000_00_0010,
+  isStrictBindingStrategy       = 0b0000_00000000_0000000000000_00_0100,
```
Comment (fkleuver) [inline packages/runtime/src/flags.ts]:
```suggestion
  isStrictBindingStrategy       = 0b00000_0000000_0000000000000_00_0100,
```

Small nit

File: packages/kernel/src/functions.ts [hunk 2]
```diff
@@ -39,12 +39,43 @@ export function isNumeric(value: unknown): value is number | string {
...
+  }
+}
+
 /**
  * Base implementation of camel and kebab cases
  */
 const baseCase = (function () {
   const enum CharKind {
-    none  = 0,
+    none = 0,
```
Comment (fkleuver) [inline packages/kernel/src/functions.ts]:
```suggestion
    none  = 0,
```

😢

File: .vscode/settings.json [hunk 3]
```diff
@@ -4,7 +4,10 @@
   "html.format.endWithNewline": true,
   "eslint.validate": [
     "javascript",
-    "typescript",
+    {
+      "autoFix": true,
```
Comment (fkleuver) [inline .vscode/settings.json]:
```suggestion
```

nonononononono

Comment (brandonseydel) [reply .vscode/settings.json]: It just gives you a bulb.

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://30442-121594173-gh.circle-artifacts.com/0/home/circleci/repo/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://30465-121594173-gh.circle-artifacts.com/0/home/circleci/repo/test/js-framework-benchmark/webdriver-ts-results/table.html

**Pull request merged**